src/composeappmanager.cc

```diff
@@ -530,6 +530,46 @@ std::string ComposeAppManager::getRunningAppsInfoForReport() const {
   return result;
 }
 
+Json::Value ComposeAppManager::getAppsState() const {
+  Json::Value apps_state;
+
+  try {
+    auto apps{app_engine_->getRunningAppsInfo()};
+    if (!apps.isNull()) {
+      const auto hash{sysroot()->getDeploymentHash(OSTree::Sysroot::Deployment::kCurrent)};
+      apps_state["deviceTime"] = TimeStamp::Now().ToString();
+      apps_state["ostree"] = hash;
+
+      for (Json::ValueIterator ii = apps.begin(); ii != apps.end(); ++ii) {
+        const auto app_name{ii.key().asString()};
+        const Json::Value& services{(*ii)["services"]};
+
+        (*ii)["state"] = "healthy";
+        for (Json::ValueConstIterator jj = services.begin(); jj != services.end(); ++jj) {
+          const Json::Value service{(*jj)};
+          // determine a service health based on its `health` field value if present
+          if (service["health"] == "unhealthy") {
+            // if it's unhealthy then an overall App is considered unhealthy
+            (*ii)["state"] = "unhealthy";
+            break;
+          }
+        }
+
+        if (!(*ii).isMember("uri")) {
+          // App's URI is stored in the `state` in the case of a regular Compose App engine
+          (*ii)["uri"] = "";  // TODO: figure out an app's URI
```

Comment:
are you still trying to find a way to do this or are you planning to leave it as "TODO" for this PR and do later?

Reply:
The later.

Reply:
It actually works for `compose apps` case not `restorable one`, we store compose app "state" in a file which contains an app uri.